tools/shared/src/main/scala/org/scalajs/core/tools/linker/LinkingUnit.scala

```diff
@@ -11,8 +11,8 @@ final class LinkingUnit private[linker] (
     val esLevel: ESLevel,
     val classDefs: List[LinkedClass],
     private[linker] val infosInternal: Map[String, Infos.ClassInfo],
-    val isComplete: Boolean,
-    dummyInternal: Unit
+    val moduleInitializers: List[ModuleInitializer],
```

Comment:
I'm ok with this right now, but it does not make a lot of sense: Why would we want to expose a read-only data structure?

We probably need a full API visibility rehaul before 1.0.0. Probably even a way to separate the linker API into "usage public" and "plugin public". The fields of LinkingUnit for example would only be plugin public, and have lower compatibility guarantees. Unclear though, how we apply SemVer to that (#2812).